plasma_framework/contracts/src/exits/payment/PaymentInFlightExitable.sol

```diff
@@ -310,4 +448,10 @@ contract PaymentInFlightExitable is
             ife.inputs[i] = inputTxs[i].outputs[outputIndex];
         }
     }
+
+    function setNonCanonicalChallenge(PaymentExitDataModel.InFlightExit storage ife)
+        private
+    {
+        ife.exitStartTimestamp = ife.exitStartTimestamp.setBit(255);
```

Comment:
I know this is on draft status, but an early comment here. I know this is from old design, but I found it pretty confusing to flag on timestamp var for this. 

I think we can:
1. Make `exitStartTimestamp` a `uint64` instead. And then we put a `bool` for `isCanonical` flag in the struct. They would automatically share the same slot in EVM as long as they can be packed into 32 bytes.
1. Flag it in `exitMap`. It is designed for flagging stuff already and it seems to have more space to flag. However, I personally much prefer the first solution as it is more readable.